src/v/kafka/server/handlers/list_offsets.cc

```diff
@@ -120,16 +122,18 @@ static ss::future<list_offset_partition_response> list_offsets_partition(
           } else if (timestamp == list_offsets_request::latest_timestamp) {
```

Comment:
Line 120 looks suspicious:

```          if (timestamp == list_offsets_request::earliest_timestamp) {
              return ss::make_ready_future<list_offset_partition_response>(
                list_offsets_response::make_partition(
                  ntp.tp.partition,
                  model::timestamp(-1),
   --->           partition->start_offset()));
          } else if (timestamp == list_offsets_request::latest_timestamp) {
```

maybe we should use k_partition (kafka::partition) instead of partition (cluster::partition)? cluster::partition doesn't transform the offsets

Reply:
great find